std/stdio.d

```diff
@@ -4522,70 +4521,70 @@ Initialize with a message and an error code.
     }
 }
 
-extern(C) void std_stdio_static_this()
+// Undocummented but public because std* handles are aliasing it
+ref File makeGlobal(alias handle)()
 {
-    static import core.stdc.stdio;
-    //Bind stdin, stdout, stderr
-    __gshared File.Impl stdinImpl;
-    stdinImpl.handle = core.stdc.stdio.stdin;
-    .stdin._p = &stdinImpl;
-    // stdout
-    __gshared File.Impl stdoutImpl;
-    stdoutImpl.handle = core.stdc.stdio.stdout;
-    .stdout._p = &stdoutImpl;
-    // stderr
-    __gshared File.Impl stderrImpl;
-    stderrImpl.handle = core.stdc.stdio.stderr;
-    .stderr._p = &stderrImpl;
+    static __gshared File.Impl impl;
+    static __gshared File result;
+    static shared bool initialized;
+    if (!initialized)
+    {
+        // Use double-checking because initOnce is inefficient.
```

Comment:
The problem I see here is that `initialized` is perfectly race-free. But the setting of the handle is a side effect. So is it possible one thread reads true while the handle has not yet been set?

I'm somewhat lost with the way this is done. You are protecting a boolean with a complicated set of calls using atomics and a mutex, and I don't fully understand why they are all there. The `result` value seems unprotected, as it's never loaded or stored atomically, whereas the booleans explicitly are inside `initOnce`.